storm-core/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -204,8 +204,8 @@ public static Executor mkExecutor(Map workerData, List<Long> executorId, Map<Str
                 throw Utils.wrapInRuntime(ex);
             }
         }
-        executor.init(idToTask);
 
+        executor.idToTask = idToTask;
```

Comment:
@HeartSaVioR It is better to pass  `idToTask` object as part of constructor instead of setting here. idToTask can be created before calling constructor.